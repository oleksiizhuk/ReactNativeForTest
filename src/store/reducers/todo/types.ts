import { TodoItemType } from '../../../components/Template/Todo/types'

export interface TodoState {
  data: TodoItemType[]
}
