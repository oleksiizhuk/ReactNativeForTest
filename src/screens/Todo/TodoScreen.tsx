import React from 'react'
import { TodoTemplate } from '../../components/Template/Todo/TodoTemplate'
import { useTodo } from './hooks/useTodo'

export const TodoScreen = () => {
  const todoData = useTodo()
  return <TodoTemplate {...todoData} />
}
