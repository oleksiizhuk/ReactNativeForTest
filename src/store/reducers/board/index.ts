import { createSlice } from '@reduxjs/toolkit'

export const initialState = {
  boardList: [{ id: '1', text: 'English', type: 'waiting', time: '60' }],
}

const slice = createSlice({
  name: 'board',
  initialState,
  reducers: {
    addItem: () => {},
    delete: () => {},
    update: () => {},
    updateStatus: () => {},
  },
})

// export const { addItem, deleteTodoItemById } = slice.actions

export default slice.reducer
