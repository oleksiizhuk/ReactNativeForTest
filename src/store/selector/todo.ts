import { createSelector } from 'reselect'
import { AppState } from '../types'

const selectTodo = (state: AppState) => state.todo

export const selectTodoData = createSelector(selectTodo, ({ data }) => data)
