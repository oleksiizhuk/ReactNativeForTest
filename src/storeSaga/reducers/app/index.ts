export * from '../../sagas/app/fetchUser';
export * from './app';
