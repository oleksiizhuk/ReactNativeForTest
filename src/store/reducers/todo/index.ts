import { createSlice } from '@reduxjs/toolkit'
import { TodoItemType } from '../../../components/Template/Todo/types'
import { TodoState } from './types'

export const initialState: TodoState = {
  data: [],
}

const slice = createSlice({
  name: 'auth',
  initialState: initialState,
  reducers: {
    addItem: (state, { payload }) => {
      state.data.push(payload)
    },
    deleteTodoItemById: (state, { payload: id }) => {
      state.data = state.data.filter((item) => item.id !== id)
    },
  },
})

export type AddItemPayload = {
  payload: TodoItemType
}

export const { addItem, deleteTodoItemById } = slice.actions

export default slice.reducer
