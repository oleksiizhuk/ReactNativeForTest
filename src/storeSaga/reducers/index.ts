import {app} from './app/app';

export const reducers = {
  app,
};
