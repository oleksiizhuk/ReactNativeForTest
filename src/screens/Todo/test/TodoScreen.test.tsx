import { render } from '@testing-library/react-native'
import { TodoScreen } from '../TodoScreen'
import { initialState } from '../../../store/reducers/todo'

const mockState = {
  todo: initialState,
}
jest.mock('react-redux', () => ({
  useDispatch: jest.fn(() => jest.fn()),
  useSelector: jest.fn((selector) => selector(mockState)),
}))
describe('TodoScreen', () => {
  it('Should match snapshot', () => {
    const tree = render(<TodoScreen />)
    expect(tree).toMatchSnapshot()
  })
})
