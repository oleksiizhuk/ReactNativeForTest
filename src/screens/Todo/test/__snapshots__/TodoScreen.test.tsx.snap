// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoScreen Should match snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginHorizontal": 16,
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={
      <Memo
        addTodoItem={[Function]}
        onChange={[Function]}
        value=""
      />
    }
    data={[]}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        collapsable={false}
        onLayout={[Function]}
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                  "marginVertical": 5,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                {
                  "backgroundColor": "#fff",
                  "borderColor": "#2860b4",
                  "borderRadius": 8,
                  "borderWidth": 1.5,
                }
              }
            >
              <TextInput
                keyboardType="default"
                onChangeText={[Function]}
                style={
                  [
                    {
                      "color": "#333",
                      "fontSize": 14,
                      "padding": 12,
                    },
                    undefined,
                  ]
                }
                value=""
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": true,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#eee",
                "borderColor": "#aaa",
                "borderRadius": 30,
                "borderWidth": 1,
                "elevation": 0,
                "opacity": 1,
                "padding": 12,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 140,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#004aad",
                    "fontSize": 16,
                    "lineHeight": 20,
                    "textAlign": "center",
                  },
                  {
                    "color": "#aaa",
                  },
                ]
              }
            >
              Add Todo
            </Text>
          </View>
        </View>
        <View>
          <Text>
            Filter:
          </Text>
        </View>
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#004aad",
                "borderRadius": 30,
                "borderWidth": 1,
                "elevation": 5,
                "opacity": 1,
                "padding": 12,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 140,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#004aad",
                    "fontSize": 16,
                    "lineHeight": 20,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Undo
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "borderColor": "#004aad",
                "borderRadius": 30,
                "borderWidth": 1,
                "elevation": 5,
                "opacity": 1,
                "padding": 12,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
                "width": 140,
              }
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#004aad",
                    "fontSize": 16,
                    "lineHeight": 20,
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              Redo
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
